src/main/java/org/craftercms/deployer/impl/TargetServiceImpl.java

```diff
@@ -91,22 +90,22 @@
     public static final String TARGET_SITE_NAME_MODEL_KEY = "site_name";
     public static final String TARGET_ID_MODEL_KEY = "target_id";
 
-    protected File targetConfigFolder;
-    protected Resource baseTargetYamlConfigResource;
-    protected Resource baseTargetYamlConfigOverrideResource;
-    protected Resource baseTargetContextResource;
-    protected Resource baseTargetContextOverrideResource;
-    protected String defaultTargetConfigTemplateName;
-    protected Handlebars targetConfigTemplateEngine;
-    protected ApplicationContext mainApplicationContext;
-    protected DeploymentPipelineFactory deploymentPipelineFactory;
-    protected TaskScheduler taskScheduler;
-    protected ExecutorService taskExecutor;
-    protected ProcessedCommitsStore processedCommitsStore;
-    protected TargetLifecycleHooksResolver targetLifecycleHooksResolver;
-    protected EncryptionAwareConfigurationReader configurationReader;
-    protected UpgradeManager<Target> upgradeManager;
-    protected Set<Target> currentTargets;
+    protected final File targetConfigFolder;
```

Comment:
Why `final`?

Reply:
The IDE automatically adds it for properties used in constructors, to prevent child classes from changing them